app/src/androidTest/java/ch/epfl/balelecbud/pointOfInterest/PointOfInterestUtilsTest.java

```diff
@@ -3,90 +3,86 @@
 import com.google.common.collect.Lists;
 import com.google.firebase.firestore.GeoPoint;
 
-import org.junit.BeforeClass;
+import org.junit.Before;
 import org.junit.Test;
 
-import java.util.LinkedList;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 
+import ch.epfl.balelecbud.BalelecbudApplication;
 import ch.epfl.balelecbud.models.Location;
-import ch.epfl.balelecbud.pointOfInterest.PointOfInterest;
 import ch.epfl.balelecbud.util.database.DatabaseWrapper;
 import ch.epfl.balelecbud.util.database.MockDatabaseWrapper;
 import ch.epfl.balelecbud.util.database.MyQuery;
 
 import static junit.framework.TestCase.assertEquals;
 
-
 public class PointOfInterestUtilsTest {
 
-    private static List<Location> locations;
-    private static MockDatabaseWrapper db;
-    private static Location l1 = new Location(1, 1);
-    private static Location l2 = new Location(1, 2);
-    private static Location l3 = new Location(2, 1);
-    private static Location l4 = new Location(2, 2);
-
-    @BeforeClass
-    public static void setup(){
-        locations = Lists.newArrayList(l1, l2, l3, l4);
-        db = (MockDatabaseWrapper) MockDatabaseWrapper.getInstance();
+    private final MockDatabaseWrapper db = MockDatabaseWrapper.getInstance();;
+    private final static Location l1 = new Location(1, 1);
+    private final static Location l2 = new Location(1, 2);
+    private final static Location l3 = new Location(2, 1);
+    private final static Location l4 = new Location(2, 2);
+
+    @Before
+    public void setup(){
+        List<Location> locations = Lists.newArrayList(l1, l2, l3, l4);
         db.resetDocument(DatabaseWrapper.LOCATIONS_PATH);
         int i = 0;
         for(Object obj : locations){
             db.storeDocumentWithID(DatabaseWrapper.LOCATIONS_PATH, Integer.toString(i++), obj);
         }
     }
 
-    private List<Location> getResList(double value, MyQuery.WhereClause.Operator op, String fieldName) throws InterruptedException, ExecutionException {
+    private List<Location> getResList(double value, MyQuery.WhereClause.Operator op, String fieldName)
+            throws InterruptedException, ExecutionException {
         MyQuery.WhereClause clause = new MyQuery.WhereClause(fieldName, op, value);
         MyQuery query = new MyQuery(DatabaseWrapper.LOCATIONS_PATH, Lists.newArrayList(clause));
         CompletableFuture<List<Location>> result = db.query(query, Location.class);
         return result.get();
     }
 
-    @Test
-    public void queryFiltersEqualsThan() throws ExecutionException, InterruptedException {
-        List<Location> resList = getResList(2, MyQuery.WhereClause.Operator.EQUAL, "longitude");
+    private void checkFilter(double i, MyQuery.WhereClause.Operator equal, Location l2, Location l4) throws InterruptedException, ExecutionException {
```

Comment:
I would at least change variable names from l2 and l4, and possibly consider passing a list. But then we should use hamcrest to make sure order does not matter

Reply:
Indeed, Android Studio created this function but I haven't really looked at it afterward. I will change that, thanks for pointing this out :-)